src/inspect_evals/swe_bench/README.md

```diff
@@ -0,0 +1,79 @@
+
+## Usage
+
+### Running the benchmark
+``swe_bench.py``` file contains ```swe_bench``` function, which creates an instance of a SWE-bench Task:
+
+```python
+from inspect_ai import eval
+from inspect_ai.solver import system_message, basic_agent
+from inspect_ai.tool import bash, python
+from inspect_evals.swe_bench import swe_bench
+
+# Create an agent that only uses bash, and an agent that only uses python
+agent = basic_agent(tools=[bash(timeout=30)])
+task = swe_bench(
+    dataset="princeton-nlp/SWE-bench_Verified",
+    solver=agent,
+    instance_ids=["sympy__sympy-22914", "sympy__sympy-13878"],
+)
+# Compare how these two agents perform.
+eval(task, model="openai/gpt-4o")
+
+```
+
+NOTE: SWE-bench will take a while to run, and uses a lot of tokens. If things are too slow, you should increase the level of parallelism - see https://inspect.ai-safety-institute.org.uk/parallelism.html. Note that running too many docker containers on your machine can also cause issues, most notably with a 'ALL PREDEFINED ADDRESS POOLS HAVE BEEN FULLY SUBNETTED' error - we don't recommend running more than 32 containers at any one time.
+
+### Comparing to official swe-bench baselines
+Submissions to [the official SWE-bench leaderboard](https://www.swebench.com/) often come with a set of trajectories and per-swebench-instance results. To download these baselines, you can git clone the official baselines repository, and copy the baselines to a local directory:
+```bash
+# Download the swe-bench baselines. NOTE: THEY ARE QUITE LARGE. 
+git clone https://github.com/swe-bench/experiments /tmp/swebench_baselines
+# Copy to current directory
+mkdir -p swebench_baselines
+cp -r /tmp/swebench_baselines/evaluation/verified/20240620_sweagent_claude3.5sonnet ./swebench_baselines/
+# Can add other files in /tmp/swebenchbaselines/evaluations/... here
+```
+
+This means that you can compare the performance of your agent not just on the full dataset, but also on subsets of that dataset. This is often useful when trying to run smaller, cheaper experiments. To make this comparison easy, we provide a scorer that will calculate the average performance of an existing baseline on that set of trajectories:
+
+```python
+from inspect_evals.swe_bench import swebench_baseline_scorer
+agent = basic_agent(tools=[bash(timeout=30)])
+scorers = [
+    swebench_baseline_scorer(
+        "./swebench_baselines/20240620_sweagent_claude3.5sonnet",
+        name="SWE-agent_3.5_sonnet",
+    ),
+    swebench_scorer(),
+]
+task = swe_bench(
+    dataset="princeton-nlp/SWE-bench_Verified",
+    solver=agent,
+    instance_ids=["sympy__sympy-22914", "sympy__sympy-13878"],
+    scorer=scorers,
+)
+
+eval(task, model="openai/gpt-4o")
+```
+
+This will lead to both numbers being reported in the final output, allowing you to compare baselines:
+
+![SWE-bench baseline comparison](./docs/swebench_comparison.jpeg)
+
+
+
```

Comment:
I would love a check of this README, to see if its in line with what you imagine as best practice. Maybe try and run the examples youreself.